frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -287,17 +243,24 @@ export const funnelLogic = kea<funnelLogicType>({
                 clearFunnel: (state) => ({ new_entity: state.new_entity }),
             },
         ],
+        exclusionFilters: [
+            (props.exclusionFilters || { type: EntityTypes.EVENTS }) as FilterType,
+            {
+                setEventExclusionFilters: (state, { filters }) => ({ ...state, ...filters, type: EntityTypes.EVENTS }),
+                setOneEventExclusionFilter: (state, { eventFilter, index }) => ({
+                    ...state,
+                    events: state.events ? state.events.map((e, e_i) => (e_i === index ? eventFilter : e)) : [],
+                }),
+            },
+        ],
```

Comment:
Wondering if we could do without this reducer, and just take the `exclusionFilters` directly from the filters? For good data hygiene, we should always have one source of truth, and derive whatever we need from there. In this case deriving this from the filters via selectors. Or is there some timing/recalculation issue causing problems?

Reply:
That's a really good idea.  Initially I added it because it looked like `<ActionFilter>` was specifically built to act on the `events` property of `filters`, but if I had a selector and action to read and write from the `filters` source of truth than that would make for a lot cleaner less buggier code.